File: src/main/java/team4/footwithme/member/oauth2/response/MemberOAuthResponse.java
package team4.footwithme.member.oauth2.response;

import team4.footwithme.member.domain.LoginProvider;
import team4.footwithme.member.domain.Member;

public record MemberOAuthResponse(
        LoginProvider provider,
        String snsId,
        String email
) {
    public static MemberOAuthResponse from(Member member){
        return new MemberOAuthResponse(
                member.getLoginType().getLoginProvider(),
                member.getLoginType().getSnsId(),
                member.getEmail()
        );
    }
}

File: src/main/java/team4/footwithme/member/oauth2/response/MemberOAuthResponse.kt
package team4.footwithme.member.oauth2.response

import team4.footwithme.member.domain.LoginProvider
import team4.footwithme.member.domain.Member

@JvmRecord
data class MemberOAuthResponse(
    val provider: LoginProvider?,
    val snsId: String?,
    val email: String?
) {
    companion object {
        fun from(member: Member?): MemberOAuthResponse {
            return MemberOAuthResponse(
                member!!.loginType!!.loginProvider,
                member.loginType!!.snsId,
                member.email
            )
        }
    }
}
